python-test-samples/step-functions-local-lambda/README.md

```diff
@@ -0,0 +1,259 @@
+
+2. **State Machine Creation**: The `fixture_sfn_client` fixture creates a Boto3 client for Step Functions and creates the state machine using the definition from `local_testing.asl.json`.
+
+3. **Test Execution**: When the test runs, it calls `execute_stepfunction` to start an execution of the state machine and monitor it until completion.
+
+4. **Validation**: The test verifies that:
+   - The execution completed successfully
+   - The HelloWorld state was properly exited
+
+5. **Cleanup**: After the test completes, the container is automatically shut down by the finalizer added in the `fixture_container`.
+
+---
+
+## Testing Workflows
+
+### Setup Docker Environment
+
+> Make sure Docker engine is running before running the tests.
+
+```shell
+step-functions-local-lambda$ docker version
+Client: Docker Engine - Community
+ Version:           24.0.6
+ API version:       1.43
+(...)
+```
+
+### Run the Unit Test - End to end python test
+
+> Set up the python environment
+
+``` shell
+step-functions-local-helloworld$ cd tests
+export AWS_ACCESS_KEY_ID='DUMMYIDEXAMPLE'
+export AWS_SECRET_ACCESS_KEY='DUMMYEXAMPLEKEY'
+export AWS_REGION='us-east-1'
+python3 -m venv venv
+source venv/bin/activate
+pip install --upgrade pip
+pip install -r requirements.txt
+```
+
+#### Run the unit tests
+
+``` shell
+step-functions-local-helloworld/tests$
+python3 -m pytest -s unit/src/test_step_functions_local.py
+```
+
+Expected output:
+``` shell
+step-functions-local-helloworld/tests$ python3 -m pytest -s unit/src/test_step_functions_local.py
+========================================================== test session starts ==========================================================
+platform linux -- Python 3.10.12, pytest-8.4.1, pluggy-1.6.0
+rootdir: /home/ubuntu/environment/serverless-test-samples-approach-2-pytest-distributed_Try3/step-functions-local-helloworld--OK/tests
+collected 1 item                                                                                                                                                                                                                                         
+
+unit/src/test_step_functions_local.py Pulling image testcontainers/ryuk:0.8.1
+Container started: ecdd9649b97d
+Waiting for container <Container: ecdd9649b97d> with image testcontainers/ryuk:0.8.1 to be ready ...
+Pulling image amazon/aws-stepfunctions-local
+Container started: d1bd33969e6b
+Waiting for container <Container: d1bd33969e6b> with image amazon/aws-stepfunctions-local to be ready ...
+.
+
+========================================================== 1 passed in 10.57s ==========================================================
+(
+```
+
+#### Clean up section
+
+> clean pyenv environment
+
+```sh
+step-functions-local-helloworld/tests$
+deactivate
+rm -rf venv/
+```
+
+> unsetting variables
+
+```sh
+unset AWS_ACCESS_KEY_ID='DUMMYIDEXAMPLE'
```

Comment:
`unset` commands should just be the variable name, and not include `='DUMMYIDEXAMPLE'`